CAIPs/caip-285.md

```diff
@@ -0,0 +1,131 @@
+---
+caip: 285
+title: JSON-RPC Method for Revoking Session Authorizations
+author: [Alex Donesky] (@adonesky1)
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/285/files
+status: Draft
+type: Standard
+created: 2024-07-12
+requires: 25, 217
+---
+
+## Simple Summary
+
+CAIP-285 introduces the `wallet_revokeSession` method for fully revoking the authorizations and properties of an active [CAIP-25][] session.
+
+## Abstract
+
+This proposal aims to enhance session management for [CAIP-25][] initiated sessions by defining a new JSON-RPC method for revoking sessions. This method provides an explicit protocol for revoking sessions with or without `sessionId`s.
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of [CAIP-25][] initated sessions by enabling the revocation of session authorizations without `sessionId`s, which don't map well to extension-based wallet's dapp connections and could add unnecessary constraints and burdens to existing flows. The proposed method provides an intuitive way to revoke authorizations of an active session, simplifying the management of session lifecycles.
+
+## Specification
+
+### Definition
+
+The `wallet_revokeSession` method revokes the entire active session.
+If a `sessionId` parameter is provided, it revokes that specific session only;
+if no `sessionId` parameter is provided and there is an active session without a `sessionId` this session gets revoked and a success result is returned;
+otherwise, an appropriate error message is sent.
+
+**Parameters:**
+
+- `sessionId` (string, optional): The session identifier.
+
+### Request
+
+The caller would interface with a wallet via the same channel by which it called `wallet_createSession` to revoke a session by calling the following JSON-RPC request:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_revokeSession",
+  "params": {}
+}
+```
+
+### Response
+
+The wallet can respond to this method with either a success result or an error message.
+
+### Success
+
+Upon a successful `wallet_revokeSession` call a wallet should remove authorizations and session properties associated with the revoked session.
+
+An example of a successful response follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": true
+}
+```
+
+### Failure States
+
+The response MUST NOT be a JSON-RPC success result in any of the following failure states.
+
+#### Generic Failure Code
+
+Unless the dapp is known to the wallet and trusted, the generic/undefined error response:
```

Comment:
Should be further defined in subsequent PRs what the threshold for trusted is because otherwise this leaves this part of the spec quite ambiguous for a wallet to implement

Reply:
agreed, this has been a TODO that's been nagging on me since Kyle first recommended doing (as Aaron recommended on the more recent call) as little error-handling as possible 😅 .  i opened a tracking issue just now so it can nag everyone, not just me.